simple_api/object/validators.py

```diff
@@ -0,0 +1,73 @@
+from inspect import isclass
+
+
+def build_validation_fn(action_validators, field_validators):
+    """
+    Prepares function for validation of input values and the action overall, instantiates validators which are not
+    already instantiated. Action_validators can be a sigle validator, Field_validators a List of tuples
+    (string, (validators...)) as prepared in object/actions.py
+    """
+    if action_validators and field_validators is None:
+        return None
+
+    if not isinstance(action_validators, (list, tuple)):
+        action_validators = action_validators,
+
+    # Validators instantiation if they are not already
+    instantiated_validators = []
+    for cls_or_inst in action_validators:
+        if isclass(cls_or_inst):
+            instantiated_validators.append(cls_or_inst())
+        else:
+            instantiated_validators.append(cls_or_inst)
+    for tup_of_validators in field_validators:
+        for validator in tup_of_validators[1]:
+            if isclass(validator):
+                instantiated_validators.append((tup_of_validators[0], validator()))
+            else:
+                instantiated_validators.append((tup_of_validators[0], validator))
+
+    def fn(**kwargs):
+        for valid in instantiated_validators:
+            if isinstance(valid, tuple):
+                if not valid[1].is_valid(valid[0], **kwargs):
+                    raise ValueError(valid[1].error_message(**kwargs))
+            else:
+                if not valid.validation_statement(**kwargs):
+                    raise ValueError(valid.error_message(**kwargs))
```

Comment:
presne proto chces slovnik - z tech jednicek a nul moc chytry nejsem :)

`isinstance` fuj, validatory nemajici jednotne API fuj (to chce sjednotit, ve forcyklu projedes a pokazde reknes pouze `validator.is_valid(**kwargs)`)

Reply:
Vyřešeno dle konverzace